### PR TITLE
[SD-4856] - Unable to unlock other users' draft

### PR DESCRIPTION
**Pull request by akintolga**
- `draft` is the "version 0" state of the story and currently every users draft is visible to everyone.
- Ideally `draft` stories should have been limited to the owners only. 
- But that change may affect other areas so we're limiting the actions on `draft` stories so that a user cannot do anything on another users draft story

### CODE REVIEW

**Comment by mdhaman** [comment]
:+1:

File: scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -3206,7 +3213,18 @@
          */
         this.edit = function(item, action) {
             if (item) {
-                authoringOpen(item._id, action || 'edit', item._type || null);
+                var actions = authoring.itemActions(item);
+                if (action) {
```
Comment (gbbr) [inline scripts/superdesk-authoring/authoring.js]: You can use `if (action && actions[action])` here.

File: scripts/superdesk-authoring/authoring.js [hunk 2]
```diff
@@ -3206,7 +3213,18 @@
...
+                    // action is provided, check if it is permitted
+                    if (actions[action]) {
+                        authoringOpen(item._id, action, item._type || null);
+                    }
+                } else {
```
Comment (gbbr) [inline scripts/superdesk-authoring/authoring.js]: Here you could do `else if` instead of nesting.

File: scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -3206,7 +3213,18 @@
          */
         this.edit = function(item, action) {
             if (item) {
```
Comment (gbbr) [inline scripts/superdesk-authoring/authoring.js]: Is there ever a case when no `item` is provided for editing? If no, then the condition is not needed imo.

**Comment by gbbr** [comment]
Looks good besides minor style comments.

**Comment by akintolga** [comment]
Closing this PR as there's a new PR to solve this problem by not displaying other users' draft stories...

**Pull request closed without merging**